SMP compat in `master`

```
[ 4 of 11] Compiling Dhall.Core       ( src/Dhall/Core.hs, dist/build/Dhall/Core.o )

src/Dhall/Core.hs:471:10: error:
    • No instance for (Semigroup (Chunks s a))
        arising from the superclasses of an instance declaration
    • In the instance declaration for ‘Monoid (Chunks s a)’
    |
471 | instance Monoid (Chunks s a) where
    |          ^^^^^^^^^^^^^^^^^^^
```

--- a/src/Dhall/Core.hs
+++ b/src/Dhall/Core.hs
@@ -471,10 +471,18 @@ data Chunks s a = Chunks [(Builder, Expr s a)] Builder
 instance Monoid (Chunks s a) where
     mempty = Chunks [] mempty
 
+#if MIN_VERSION_base(4,11,0)
+instance Semigroup (Chunks s a) where
+    (<>) (Chunks xysL zL) (Chunks         []    zR) =
+        Chunks xysL (zL <> zR)
+    (<>) (Chunks xysL zL) (Chunks ((x, y):xysR) zR) =
+        Chunks (xysL ++ (zL <> x, y):xysR) zR
+#else
     mappend (Chunks xysL zL) (Chunks         []    zR) =
         Chunks xysL (zL <> zR)
     mappend (Chunks xysL zL) (Chunks ((x, y):xysR) zR) =
         Chunks (xysL ++ (zL <> x, y):xysR) zR
+#endif
 
 instance IsString (Chunks s a) where
     fromString str = Chunks [] (fromString str)

--- a/src/Dhall/Parser.hs
+++ b/src/Dhall/Parser.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP               #-}
 {-# LANGUAGE DeriveDataTypeable         #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE OverloadedStrings          #-}
@@ -97,7 +98,12 @@ newtype Parser a = Parser { unParser :: Text.Trifecta.Parser a }
 instance Monoid a => Monoid (Parser a) where
     mempty = pure mempty
 
+#if MIN_VERSION_base(4,11,0)
+instance Semigroup a => Semigroup (Parser a) where
+    (<>) = liftA2 (<>)
+#else
     mappend = liftA2 mappend
+#endif
 
 instance IsString a => IsString (Parser a) where
     fromString x = fmap fromString (Text.Parser.Char.string x)

